qa/rpc-tests/checkblockatheight.py

```diff
@@ -52,33 +50,41 @@ def setup_network(self, split=False, minAge=FINALITY_MIN_AGE):
                 ["-debug=py", "-debug=cbh", "-cbhsafedepth="+str(FINALITY_SAFE_DEPTH), "-cbhminage="+str(minAge)]
             ])
 
-        if not split:
-            # 2 and 3 are joint only if split==false
-            connect_nodes_bi(self.nodes, 2, 3)
-            connect_nodes_bi(self.nodes, 3, 2)
-            sync_blocks(self.nodes[2:NUMB_OF_NODES])
-            sync_mempools(self.nodes[2:NUMB_OF_NODES])
-
-        connect_nodes_bi(self.nodes, 0, 1)
-        connect_nodes_bi(self.nodes, 1, 0)
-        connect_nodes_bi(self.nodes, 1, 2)
-        connect_nodes_bi(self.nodes, 2, 1)
+        for idx in range(NUMB_OF_NODES - 1):
+            connect_nodes_bi(self.nodes, idx, idx + 1)
         self.is_network_split = split
         self.sync_all()
 
     def split_network(self):
-        # Split the network of 4 nodes into nodes 0-1-2 and 3.
-        assert not self.is_network_split
-        disconnect_nodes(self.nodes[2], 3)
-        disconnect_nodes(self.nodes[3], 2)
+        # Disconnect the dishonest node from the network
+        idx = DISHONEST_NODE_INDEX
+        if idx == NUMB_OF_NODES - 1:
+            disconnect_nodes(self.nodes[idx], idx - 1)
+            disconnect_nodes(self.nodes[idx - 1], idx)
+        elif idx == 0:
+            disconnect_nodes(self.nodes[idx], idx + 1)
+            disconnect_nodes(self.nodes[idx + 1], idx)
+        else:
+            disconnect_nodes(self.nodes[idx], idx - 1)
+            disconnect_nodes(self.nodes[idx - 1], idx)
+            disconnect_nodes(self.nodes[idx], idx + 1)
+            disconnect_nodes(self.nodes[idx + 1], idx)
+            connect_nodes_bi(self.nodes, idx - 1, idx + 1)
         self.is_network_split = True
 
 
     def join_network(self):
-        #Join the (previously split) network pieces together: 0-1-2-3
-        assert self.is_network_split
-        connect_nodes_bi(self.nodes, 2, 3)
-        connect_nodes_bi(self.nodes, 3, 2)
+        #Join the (previously split) network pieces together
+        idx = DISHONEST_NODE_INDEX
+        if idx == NUMB_OF_NODES - 1:
+            connect_nodes_bi(self.nodes, idx, idx - 1)
+        elif idx == 0:
+            connect_nodes_bi(self.nodes, idx, idx + 1)
+        else:
+            disconnect_nodes(self.nodes[idx - 1], idx + 1)
+            disconnect_nodes(self.nodes[idx + 1], idx - 1)
+            connect_nodes_bi(self.nodes, idx, idx - 1)
+            connect_nodes_bi(self.nodes, idx, idx + 1)
         sync_blocks(self.nodes, 1, False, 5)
```

Comment:
I'm thinking about the usage of `sync_blocks(self.nodes, 1, False, 5)` limiting wait until only 5 loops; I know it's something not directly introduced by this PR but maybe it is worth double checking if it's not better to leave unlimited wait (and rely on CI timeout).

Reply:
Yes, agree. I revised the `sync_blocks` mechanism on the whole test, and replaced the timed version with non-timed one throughout the test